### PR TITLE
remove atomic service reference

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java [hunk 1]
```diff
@@ -36,57 +36,31 @@
...
 public class CDIManagedObjectService implements ManagedObjectService {
 
-    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
-    private final AtomicServiceReference<DefaultManagedObjectService> defaultMOSRef = new AtomicServiceReference<DefaultManagedObjectService>("defaultManagedObjectService");
+    private DefaultManagedObjectService defaultMOS;
+    private CDIService cdiService;
 
     private CDIRuntime cdiRuntime;
 
     public void activate(ComponentContext cc) {
```
Comment (tjwatson) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java]: No need for an empty activate method.  You should delete this method.

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java [hunk 2]
```diff
@@ -36,57 +36,31 @@
...
-        cdiServiceRef.activate(cc);
-        defaultMOSRef.activate(cc);
     }
 
     public void deactivate(ComponentContext cc) {
-        cdiServiceRef.deactivate(cc);
-        defaultMOSRef.deactivate(cc);
         cdiRuntime = null;
```
Comment (tjwatson) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java]: There is no need to null out this mandatory reference on deactivate.  I suggest deleting this whole method.

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java [hunk 3]
```diff
@@ -36,57 +36,31 @@
                                                                                                                                                         "service.ranking:Integer=9999" })
 public class CDIManagedObjectService implements ManagedObjectService {
 
-    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
-    private final AtomicServiceReference<DefaultManagedObjectService> defaultMOSRef = new AtomicServiceReference<DefaultManagedObjectService>("defaultManagedObjectService");
+    private DefaultManagedObjectService defaultMOS;
+    private CDIService cdiService;
 
     private CDIRuntime cdiRuntime;
```
Comment (tjwatson) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java]: Make `defaultMOS` and `cdiRuntime` final, remove `cdiService`  and add an `@Activate` constructor that sets these final values with constructor injection:

```java
@Activate
public CDIManagedObjectService(@Reference CDIService cdiService,
                               @Reference DefaultManagedObjectService defaultMOS) {
    this.cdiRuntime =  (CDIRuntime) cdiService;
    this.defaultMOS = defaultMOS;
}
```

The default for reference is static and mandatory, no need to name the reference.  Keep it simple.

**Review by tjwatson — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java [hunk 4]
```diff
@@ -36,57 +36,31 @@
...
-
-    private CDIService getCDIService() {
-        return AccessController.doPrivileged(new PrivilegedAction<CDIService>() {
-            @Override
-            public CDIService run() {
-                return cdiServiceRef.getServiceWithException();
-            }
-        });
-
+    @Reference(name = "defaultManagedObjectService", service = DefaultManagedObjectService.class, cardinality = ReferenceCardinality.MANDATORY)
```
Comment (tjwatson) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java]: Forgot to mention this.  Delete the two `@Reference` set methods when using constructor injection for the references.

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java [hunk 5]
```diff
@@ -114,24 +70,24 @@ private CDIRuntime getCDIRuntime() {
...
+            return new CDIInterceptorManagedObjectFactoryImpl<T>(klass, cdiRuntime);
         } else {
-            return getDefaultManagedObjectService().createInterceptorManagedObjectFactory(mmd, klass);
+            return defaultMOS.createInterceptorManagedObjectFactory(mmd, klass);
         }
     }
 
     private boolean isCDIEnabled(ModuleMetaData mmd) {
-        return getCDIRuntime() != null && getCDIRuntime().isModuleCDIEnabled(mmd);
+        return cdiRuntime != null && cdiRuntime.isModuleCDIEnabled(mmd);
```
Comment (tjwatson) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java:80]: No need to have a check for null here.

**Review by tjwatson — APPROVED** [review]
My comment is a nit.  It would be more clear to not check for null.  I'm fine merging as is, but if you have time it would be good to remove the null check.

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Pull request merged**